File: src/_01_CreationalPattern/_01_03_AbstractFactory/Ship.js
/**
 * @typedef { import("./Anchor").default } Anchor
 * @typedef { import("./Wheel").default } Wheel
 */

export default class Ship {
  /** @type { string } */
  #name;

  /** @type { string } */
  #type;

  /** @type { string } */
  #color;

  /** @type { string } */
  #logo;

  /** @type { Anchor } */
  #anchor;

  /** @type { Wheel } */
  #wheel;

  /** @returns { string } */
  getName() {
    return this.#name;
  }
  /** @param { string } name */
  setName(name) {
    this.#name = name;
  }

  /** @returns { string } */
  getType() {
    return this.#type;
  }
  /** @param { string } type */
  setType(type) {
    this.#type = type;
  }

  /** @returns { string } */
  getColor() {
    return this.#color;
  }
  /** @param { string } color */
  setColor(color) {
    this.#color = color;
  }

  /** @returns { string } */
  getLogo() {
    return this.#logo;
  }
  /** @param { string } */
  setLogo(logo) {
    this.#logo = logo;
  }

  /** @returns { Anchor } */
  getAnchor() {
    return this.#anchor;
  }
  /** @param { Anchor } anchor */
  setAnchor(anchor) {
    this.#anchor = anchor;
  }
  
  /** @returns { Wheel } */
  getWheel() {
    return this.#wheel;
  }
  /** @param { Wheel } wheel */
  setWheel(wheel) {
    this.#wheel = wheel;
  }

  printInfo() {
    console.log(`
      Ship {
        name: ${this.#name},
        type: ${this.#type},
        color: ${this.#color},
        logo: ${this.#logo},
        anchor: ${this.#anchor.getName()},
        wheel: ${this.#wheel.getName()},
      };
    `);
  }
}

File: src/_01_CreationalPattern/_01_03_AbstractFactory/java/after/Ship.java
package _01_CreationalPattern._01_03_AbstractFactory.java.after;

public class Ship {

  private String name;
  private String type;
  private String color;
  private String logo;
  private Anchor anchor;
  private Wheel wheel;
  
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

  public String getType() {
    return type;
  }
  public void setType(String type) {
    this.type = type;
  }

  public String getColor() {
    return color;
  }
  public void setColor(String color) {
    this.color = color;
  }

  public String getLogo() {
    return logo;
  }
  public void setLogo(String logo) {
    this.logo = logo;
  }

  public Anchor getAnchor() {
    return anchor;
  }
  public void setAnchor(Anchor anchor) {
    this.anchor = anchor;
  }
  
  public Wheel getWheel() {
    return wheel;
  }
  public void setWheel(Wheel wheel) {
    this.wheel = wheel;
  }
}
